### PR TITLE
♻️ Remove role button from buttons

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Addresses #31342.

In particular to look at @gmajoulet are the changes to Web Stories components.

### CODE REVIEW

File: examples/adsense.amp.html [hunk 1]
```diff
@@ -58,7 +58,7 @@
       id="amp-user-notification">
     This site uses cookies to personalize content.
     <a href="#learn-more">Learn more.</a>
-    <button on="tap:amp-user-notification.dismiss" role="button" tabindex="0">I accept</button>
+    <button on="tap:amp-user-notification.dismiss" tabindex="0">I accept</button>
```
Comment (caroqliu) [inline examples/adsense.amp.html:61]: Buttons are also focusable, can we remove any `tabindex="0"`? (other #s are meaningful and can be left as is for providing focus order though)

File: examples/amp-consent/cmp-vendors.amp.html [hunk 2]
```diff
@@ -352,7 +352,7 @@ <h3>Image that is NOT blocked by consent</h3>
             }
           </script>
           <div id="quantcast-post-prompt">
-            <button on="tap:conset.prompt(consent=quantcast)" role="button">Privacy settings</button>
+            <button on="tap:conset.prompt(consent=quantcast)">Privacy settings</button>
```
Comment (caroqliu) [inline examples/amp-consent/cmp-vendors.amp.html:355]:
```suggestion
            <button on="tap:consent.prompt(consent=quantcast)">Privacy settings</button>
```

**Review by caroqliu — APPROVED** [review]

File: examples/adsense.amp.html (hunk 1 above)
Comment (kristoferbaxter) [reply examples/adsense.amp.html:61]: I can do this in a followup PR. This one touches a lot of files already.

**Review by Enriqe — APPROVED** [review]
`amp-story/*` approval

**Comment by gmajoulet** [comment]
This PR needs to be rollbacked, or a fix needs to be sent in the next hour.

Stories use the `role="button"` to know not to navigate to the next page on click. I believe clicking the "Play video" fallback button, or "Enable 360" button will now incorrectly navigate to the next story page.

cc @Enriqe @kristoferbaxter

**Comment by kristoferbaxter** [comment]
Feel free to roll this PR back, but I'd also ask to do the following:
1. Stop relying on `role=button` for navigation canceling, enhancing to more standard DOM semantics.
2. Add tests so those less familiar with the `amp-story` code base don't make this same change.

**Comment by Enriqe** [comment]
PR for story revert: https://github.com/ampproject/amphtml/pull/32283

**Pull request merged**